subsys/bluetooth/shell/cs.c

```diff
@@ -194,6 +194,184 @@ static int cmd_cs_test_simple(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_remove_config(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err = 0;
+	if (default_conn == NULL) {
+		shell_error(sh, "Conn handle error, at least one connection is required.");
+		return -ENOEXEC;
+	}
+
+	uint8_t config_id = strtoul(argv[1], NULL, 10);
+
+	err = bt_cs_remove_config(default_conn, config_id);
+	if (err) {
+		shell_error(sh, "bt_cs_remove_config returned error %d", err);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+
+static int cmd_create_config(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err = 0;
+	enum bt_cs_create_config_context context;
+	struct bt_cs_create_config_params params;
+
+	if (default_conn == NULL) {
+		shell_error(sh, "Conn handle error, at least one connection is required.");
+		return -ENOEXEC;
+	}
+
+	params.id = strtoul(argv[1], NULL, 10);
+	if (!strcmp(argv[2], "local-only")) {
+		context = BT_CS_CREATE_CONFIG_CONTEXT_LOCAL_ONLY;
+	} else if (!strcmp(argv[2], "local-only")) {
+		context = BT_CS_CREATE_CONFIG_CONTEXT_LOCAL_AND_REMOTE;
+	} else {
+		shell_error(sh, "Invalid context: %s", argv[2]);
+		shell_help(sh);
+		return SHELL_CMD_HELP_PRINTED;
+	}
+
+	if (!strcmp(argv[3], "initiator")) {
+		params.role = BT_CONN_LE_CS_CONFIG_ROLE_INITIATOR;
+	} else if (!strcmp(argv[3], "reflector")) {
+		params.role = BT_CONN_LE_CS_CONFIG_ROLE_REFLECTOR;
+	} else {
+		shell_error(sh, "Invalid role: %s", argv[3]);
+		shell_help(sh);
+		return SHELL_CMD_HELP_PRINTED;
+	}
+
+	/* Set the default values */
+	params.main_mode_type = BT_CONN_LE_CS_CONFIG_MAIN_MODE_2;
+	params.sub_mode_type = BT_CONN_LE_CS_CONFIG_SUB_MODE_1;
+	params.min_main_mode_steps = 0x05;
+	params.max_main_mode_steps = 0x0A;
+	params.main_mode_repetition = 0;
+	params.mode_0_steps = 1;
+	params.rtt_type = BT_CS_TEST_RTT_AA_ONLY;
+	params.cs_sync_phy = BT_CONN_LE_CS_CONFIG_CS_SYNC_2M;
+	params.channel_map_repetition = 1;
+	params.channel_selection_type = BT_CONN_LE_CS_CONFIG_CHSEL_TYPE_3B;
+	params.ch3c_shape = BT_CONN_LE_CS_CONFIG_CH3C_SHAPE_HAT;
+	params.ch3c_jump = 2;
+
+	memset(params.channel_map, 0xFF, 10);
```

Comment:
I wonder if we should expose a "nice" way of setting channel maps. I also wrote code like:

![channel map](https://github.com/user-attachments/assets/f22ec2d9-afb1-4b11-a17d-6592d5195bfb)

but it's not very readable

Reply:
I can add a function to set all the valid bits in a given chmap

Reply:
maybe also a macro similar to CS_CHANNEL_BIT() that lets you access channel X in the array